packages/compass-saved-aggregations-queries/src/stores/aggregations-queries-items.ts

```diff
@@ -1,5 +1,102 @@
-function reducer(state = {}): typeof state {
+import { remote } from 'electron';
+import { join } from 'path';
+import fs from 'fs';
+import { AnyAction } from 'redux';
+import { FavoriteQueryCollection } from '@mongodb-js/compass-query-history';
+
+import { actions } from '../actions/aggregations-queries-actions';
+interface Item {
+  id: string;
+  name: string;
+  namespace: string;
+  lastModified: number;
+}
+
+export interface Aggregation extends Item {
+  pipeline: unknown[];
+}
+
+export interface Query extends Item {
+  collation: Record<string, unknown>;
+  filter: Record<string, unknown>;
+  limit: number;
+  project: Record<string, unknown>;
+  skip: number;
+  sort: Record<string, number>;
+}
+
+export type State = {
+  loading: boolean;
+  items: Array<Aggregation | Query>;
+};
+
+const INITIAL_STATE: State = {
+  loading: true,
+  items: [],
+};
+
+function reducer(state = INITIAL_STATE, action: AnyAction): State {
+  if (action.type === actions.FETCH_DATA) {
+    return {
+      items: [...getAggregations(), ...getQueries()],
+      loading: false,
+    };
+  }
   return state;
 }
 
+const getAggregations = () => {
+  const dir = join(remote.app.getPath('userData'), 'SavedPipelines');
+  const aggregations = fs
+    .readdirSync(dir)
+    .filter((file) => file.endsWith('.json'))
+    .map((file) => {
+      const filePath = join(dir, file);
+      const data = JSON.parse(fs.readFileSync(filePath, 'utf8'));
+      data.lastModified = fs.statSync(filePath).mtimeMs;
+      return data;
+    });
+  return mapAggregations(aggregations);
+};
+
+const mapAggregations = (aggregations: unknown[]): Aggregation[] => {
+  return aggregations.map((aggregation: any) => ({
+    id: aggregation.id,
+    name: aggregation.name,
+    namespace: aggregation.namespace,
+    lastModified: aggregation.lastModified,
+
+    pipeline: aggregation.pipeline,
+  }));
+};
+
+const getQueries = (): Query[] => {
```

Comment:
suggestion: not sure how helpful this is but in connection storage I've got this function:

``` js
function promisifyAmpersandMethod<T>(
  fn: (options: AmpersandMethodOptions<T>) => void
): () => Promise<T> {
  return () =>
    new Promise((resolve, reject) => {
      fn({
        success: (model: T) => {
          resolve(model);
        },
        error: (model: T, error: Error) => {
          reject(error);
        },
      });
    });
}
```

So you could do just:

``` js
async function getQueries(): Promise<Query[]> {
    const fetch = promisifyAmpersandMethod(
       collection.fetch.bind(collection)
    );
    
    return mapQueryModels(await fetch());
}
```

Reply:
imported `promisifyAmpersandMethod` function. thanks :)